src/dbnode/persist/fs/streaming_write.go

```diff
@@ -96,27 +98,17 @@ func NewStreamingWriter(opts StreamingWriterOptions) (StreamingWriter, error) {
 	}
 	m, k := bloom.EstimateFalsePositiveRate(
 		plannedRecordsCount,
-		opts.Options.IndexBloomFilterFalsePositivePercent(),
+		w.options.IndexBloomFilterFalsePositivePercent(),
 	)
-	bloomFilter := bloom.NewBloomFilter(m, k)
+	w.bloomFilter = bloom.NewBloomFilter(m, k)
```

Comment:
This bloom filter is built per iteration, but it seems to only be created from `w.options`; can probably move this into the constructor and free it on `StreamingWriter.Finalize()`